libsolidity/interface/CompilerStack.cpp

```diff
@@ -696,18 +696,32 @@ bool CompilerStack::compile(State _stopAfter)
 					}
 				}
 	m_stackState = CompilationSuccessful;
-	this->link();
+	if (performLinking == true)
+		this->link();
 	return true;
 }
 
-void CompilerStack::link()
+list<string> CompilerStack::link()
 {
 	solAssert(m_stackState >= CompilationSuccessful, "");
+
+	std::list<std::string> libNames;
+
 	for (auto& contract: m_contracts)
 	{
-		contract.second.object.link(m_libraries);
+		for (auto& library: m_libraries)
+			for (auto& con: m_contracts)
+				if (library.first == con.first.substr(con.first.find_last_of('/') + 1))
+				{
+					contract.second.object.bytecode.push_back(static_cast<uint8_t>(solidity::evmasm::Instruction::PUSH20));
```

Comment:
What are you doing here?

Reply:
I am checking if there are any matching libraries in m_contracts and if there are any, I am adding them to the object's linkReferences. In compiler mode, before that change linkReferences where always empty and it did not even enter the linking process in LinkerObject::link().

Reply:
And why are you doing this? Note that you are adding a push instruction to the end of the bytecode after the assembly stage - I don't think this will do any good.

Reply:
I added the push instruction because I saw it was used in LinkerObject::assemble() when setting the linkReferences, and thought it was necessary. I will delete it.